OpenAPI/kiota/models.md

```diff
@@ -21,6 +21,27 @@ All models declared inline with an operation will be generated under the namespa
 
 Models in an [allOf](https://spec.openapis.org/oas/latest.html#composition-and-inheritance-polymorphism) schema declaration will inherit from each other. Where the uppermost type in the collection is the greatest ancestor of the chain.
 
+### allOf interpretation rules
+
+The following table describes how kiota will project any schema with allOf entries when processing an API description:
+
+| Number of Properties | Number of allOf entries without a reference / inline | Number of allOf entries with a reference | Total number of allOf entries | Result |
+| -------------------- | ---------------------------------------- | ------------------------------------------- | ----------------------------- | ------ |
+| 0 | 0 | 0 | 0 | Ignored/Invalid |
+| 0 | 0 or 1 | 0 or 1 | 1 | Ignored, will process the only allOf entry instead and use the original schema's description and name. |
+| 1 or more | 0 | 0 | 0 | Class/interface without a parent type. |
+| 0 | 1 | 1 | 2 | Class/interface with properties from the inline schema and a parent type from the referenced schema. |
+| 1 or more | 0 | 1 | 1 | Class/interface with properties from the current schema and a parent type from the referenced schema. |
+| 1 or more | 1 | 0 | 1 | Class/interface with properties from the current schema and a parent type from the inline schema. |
+| 0 | 0 | 1 | 1 | Class/interface with properties from the referenced schema and without a parent type. |
+| 0 | 1 | 0 | 1 | Class/interface with properties from the inline schema and without a parent type. |
+| 1 or more | 1 | 1 | 2 | Class/interface with properties from the current schema and a parent type from referenced schema if it has properties. |
```

Comment:
Does the inlined allof schema get ignored?

Reply:
yes

Reply:
This comment should qualify that it assumes the inline schema has no properties and is therefore ignored.  Currently the code does not check for that.

Reply:
and implement this fix https://github.com/microsoft/kiota/issues/4714